File: versioned_docs/version-1.19.x/examples/go/src/data_management_rename_entries.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Get or create a bucket with the name "my-bucket"
	bucket, err := client.CreateOrGetBucket(context.Background(), "example-bucket", nil)
	if err != nil {
		panic(err)
	}

	// Rename the entry "entry_1" to "entry_2"
	err = bucket.RenameEntry(context.Background(), "entry_1", "entry_2")
	if err != nil {
		panic(err)
	}

	// Check that the entry was renamed
	entries, err := bucket.GetEntries(context.Background())
	if err != nil {
		panic(err)
	}
	entryNames := make([]string, len(entries))
	for i, entry := range entries {
		entryNames[i] = entry.Name
	}
}

File: versioned_docs/version-1.19.x/examples/py/src/data_management_rename_entries.py
from reduct import Client


async def rename_entry():
    # Create a client with the base URL and API token
    async with Client("http://127.0.0.1:8383", api_token="my-token") as client:

        # Get the bucket with the name "example-bucket"
        bucket = await client.get_bucket("example-bucket")

        # Rename the entry "entry_1" to "entry_2"
        await bucket.rename_entry("entry_1", "entry_2")

        # Check that the entry was renamed
        entries = await bucket.get_entry_list()
        entry_names = [entry.name for entry in entries]
        assert "entry_2" in entry_names
        assert "entry_1" not in entry_names


if __name__ == "__main__":
    import asyncio

    asyncio.run(rename_entry())
